src/components/engine/engine-instances-table.tsx

```diff
@@ -397,36 +410,124 @@ const RemoveModal = ({
     }
   };
 
+  const onClickRequestToCancel = async () => {
```

Comment:
throw this in a `useMutation` please - that way you don't get double submissions (u can use the `isLoading` from the return to disable the button as well / show a spinner and it will handle error cases better

I also spy the same thing on the `onClickRemove` above so please do it there, too

Reply:
same here, just move the success / error case to the mutate options callbacks `onSuccess` and `onError`